tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/experiment_builder.py

```diff
@@ -546,3 +763,46 @@ def reopen_experiment_builders(
 
         for eb in experiment_builders:
             yield eb
+
+
+def _write_X_normalized(args: Tuple[str, int, int, npt.NDArray[np.float32]]) -> None:
+    """
+    Helper for ExperimentBuilder.write_X_normalized.
+
+    Read indicated rows from X['raw'], write to X['normalized']
+    """
+    experiment_uri, obs_joinid_start, n, raw_sum = args
+    logging.info(f"Write X normalized - starting block {obs_joinid_start}")
+    with soma.open(
+        urlcat(experiment_uri, "ms", MEASUREMENT_RNA_NAME, "X", "raw"), mode="r", context=SOMA_TileDB_Context()
+    ) as X_raw:
+        with soma.open(
+            urlcat(experiment_uri, "ms", MEASUREMENT_RNA_NAME, "X", "normalized"),
+            mode="w",
+            context=SOMA_TileDB_Context(),
+        ) as X_normalized:
+            with create_thread_pool_executor(max_workers=8) as pool:
+                lazy_reader = EagerIterator(
+                    X_raw.read(coords=(slice(obs_joinid_start, obs_joinid_start + n - 1),)).tables(),
+                    pool=pool,
+                )
+                lazy_divider = EagerIterator(
+                    (
+                        (
+                            X_tbl["soma_dim_0"],
+                            X_tbl["soma_dim_1"],
+                            X_tbl["soma_data"].to_numpy() / raw_sum[X_tbl["soma_dim_0"]],
+                        )
+                        for X_tbl in lazy_reader
+                    ),
+                    pool=pool,
+                )
+                for soma_dim_0, soma_dim_1, soma_data in lazy_divider:
+                    X_normalized.write(
+                        pa.Table.from_arrays(
+                            [soma_dim_0, soma_dim_1, soma_data],
+                            names=["soma_dim_0", "soma_dim_1", "soma_data"],
+                        )
+                    )
```

Comment:
Could this be even faster if the divide and write operations were also performed in parallel? Is the EagerIterator always active or does it end up being idle, waiting for the downstream operations to complete?